### PR TITLE
journal: fix error handling when compressing journal objects

### PR DESCRIPTION
**Pull request by poettering**
Let's make sure we handle compression errors properly, and don't
misunderstand an error for success.

Also, let's actually compress things if lz4 is enabled.

Fixes #1662.

### CODE REVIEW

**Comment by poettering** [comment]
@vcaputo, @keszybz this is a reworked fix for #1662. Please have a look.

File: src/journal/journal-file.c [hunk 1]
```diff
@@ -1076,23 +1076,24 @@ static int journal_file_append_data(
...
 #if defined(HAVE_XZ) || defined(HAVE_LZ4)
-        if (f->compress_xz &&
-            size >= COMPRESSION_SIZE_THRESHOLD) {
+        if (JOURNAL_FILE_COMPRESS(f) && size >= COMPRESSION_SIZE_THRESHOLD) {
                 size_t rsize = 0;
 
                 compression = compress_blob(data, size, o->data.payload, &rsize);
 
-                if (compression) {
+                if (compression >= 0) {
```
Comment (crawford) [inline src/journal/journal-file.c:1084]: Why not strictly greater than? This is checking against the bitfield containing OBJECT_COMPRESSED_LZ4 and OBJECT_COMPRESSED_XZ, right?

Comment (poettering) [inline src/journal/journal-file.c:1084]: Well, it's customary in systemd to indicate success with return values >= 0, and errors with return values < 0... Now, it's true that the two OBJECT_COMPRESSED_XYZ values are actually != 0, so we could write > 0, but I mean they actually aren't 3 or 4 either, so I'd rather not get into comparing with too specific values, but stick to the simple decision between "is this an error" or "is this not an error"... Hope that makes some sense...

Comment (crawford) [inline src/journal/journal-file.c:1084]: Yes, I actually misread `o->object.flags |= compression;` and thought that it was always setting that to be non-zero (which wouldn't make sense of `compress_blob()` returned 0). This makes sense.

**Comment by keszybz** [comment]
Tested with:

```
src/journal-remote/log-generator.py --data-size=200000 --data-type=simple 1000 > input
./systemd-journal-remote -o output.journal --split-mode=none input
```

**Pull request merged**